### PR TITLE
Convert most remaining C-style casts to C++-style ones

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/PlatformEnvironment.cpp [hunk 1]
```diff
@@ -124,35 +124,38 @@ std::unique_ptr<IPlatformEnvironment> OpenRCT2::CreatePlatformEnvironment()
 
     // Set default paths
     std::string basePaths[DIRBASE_COUNT];
-    basePaths[(size_t)DIRBASE::OPENRCT2] = Platform::GetInstallPath();
-    basePaths[(size_t)DIRBASE::USER] = Path::Combine(Platform::GetFolderPath(SPECIAL_FOLDER::USER_DATA), subDirectory);
-    basePaths[(size_t)DIRBASE::CONFIG] = Path::Combine(Platform::GetFolderPath(SPECIAL_FOLDER::USER_CONFIG), subDirectory);
-    basePaths[(size_t)DIRBASE::CACHE] = Path::Combine(Platform::GetFolderPath(SPECIAL_FOLDER::USER_CACHE), subDirectory);
-    basePaths[(size_t)DIRBASE::DOCUMENTATION] = Platform::GetDocsPath();
+    basePaths[static_cast<size_t>(DIRBASE::OPENRCT2)] = Platform::GetInstallPath();
```
Comment (AaronVanGeffen) [inline src/openrct2/PlatformEnvironment.cpp:127]: This does not look like an improvement, or even desirable. Perhaps basePaths should just be a `std::map` instead?

Comment (Gymnasiast) [reply src/openrct2/PlatformEnvironment.cpp:127]: Can't say I really agree. The point is to move away from C-style casts to more clearly defined behaviour, not to make it look pretty.

That said, you're welcome to push any improvements to this PR.

Comment (tupaschoal) [reply src/openrct2/PlatformEnvironment.cpp:127]: I'm with @Gymnasiast here, we can change that afterwards, it's a bit out of scope in here

File: src/openrct2/drawing/Drawing.String.cpp [hunk 2]
```diff
@@ -313,7 +313,7 @@ void draw_string_centred_raw(rct_drawpixelinfo* dpi, int32_t x, int32_t y, int32
 {
     ScreenCoordsXY screenCoords(dpi->x, dpi->y);
     gCurrentFontSpriteBase = FONT_SPRITE_BASE_MEDIUM;
-    gfx_draw_string(dpi, (char*)"", COLOUR_BLACK, screenCoords);
+    gfx_draw_string(dpi, const_cast<char*>(""), COLOUR_BLACK, screenCoords);
```
Comment (ZehMatt) [inline src/openrct2/drawing/Drawing.String.cpp:316]: I dont think a cast is required, passing ```""``` should be of type ```const char*```

Comment (duncanspumpkin) [reply src/openrct2/drawing/Drawing.String.cpp:316]: Its removing the const I believe. Really all of the draw_strings need to be modified to take const.

Comment (ZehMatt) [reply src/openrct2/drawing/Drawing.String.cpp:316]: I looked at it, it already takes ```const_utf8string```

**Pull request merged**